sdk/core/azure-core/src/main/java/com/azure/core/util/polling/SyncOperationResourcePollingStrategy.java

```diff
@@ -219,6 +219,7 @@ public U getResult(PollingContext<T> pollingContext, TypeReference<U> resultType
         HttpRequest request = new HttpRequest(HttpMethod.GET, finalGetUrl);
         try (HttpResponse response = httpPipeline.sendSync(request, context)) {
             BinaryData responseBody = response.getBodyAsBinaryData();
+            responseBody.toString();
             return PollingUtils.deserializeResponseSync(responseBody, serializer, resultType);
         }
```

Comment:
Can we remove the try-with-resources here and not eagerly close `HttpResponse`?

Reply:
Removing it may cause resource leak.
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/NettyAsyncHttpResponse.java#L98-L101
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpResponseSync.java#L87-L97

As an LRO, user likely won't have a chance to close the `HttpRequest` by themselves.

Agree that stream would be bad (e.g. when the LRO results in a downloadable file). Hence we likely don't want this `toString`. Will need further discussion.

Reply:
Instead of using the logic `BinaryData.toString()` to ensure that the `BinaryData` is durable for its lifetime could we add logic into the `PollingUtils.deserializeResponse` methods where it inspects the replayability of the `BinaryData`?

```java
public static <T> T deserializeResponseSync(BinaryData binaryData, ObjectSerializer serializer,
    TypeReference<T> typeReference) {
    if (TypeUtil.isTypeOrSubTypeOf(BinaryData.class, typeReference.getJavaType())) {
        return (T) binaryData;
    } else {
        return binaryData.toObject(typeReference, serializer);
    }
}
```

becomes

```java
public static <T> T deserializeResponseSync(BinaryData binaryData, ObjectSerializer serializer,
    TypeReference<T> typeReference) {
    if (TypeUtil.isTypeOrSubTypeOf(BinaryData.class, typeReference.getJavaType())) {
        return (T) binaryData.toReplayableBinaryData();
    } else {
        return binaryData.toObject(typeReference, serializer);
    }
}
```

This should ensure that the `BinaryData` is replayable for its lifetime without the overhead of `toString`. 

This could be a nice stopgap as I believe the larger issue is that different underlying HTTP stacks handle differently. Ex, Netty by default uses shared memory pools to reduce allocation which has the downside that once the response is closed Netty reclaims the memory segments used by the response, but other HTTP stacks may have the memory read into memory.

Reply:
Interestingly, `toReplayableBinaryData()` not work (I too thought it would).
Haven't dig deep on the reason why it won't work.

My concern was more on resource close/leak.

E.g. if we have a real return of BinaryData represents an octet-stream. User may actually choose not to read all of it (maybe they only need part of the data). Then when do we close the request and return the connection to HttpClient connection pool.

Reply:
I took a look into the `toReplayableBinaryData(Async)` methods and the issue is that the `Flux<ByteBuffer>` is lazily copied on access of the `BinaryData` which means that the memory may be reclaimed after the `HttpResponse` is closed but before the `Flux<ByteBuffer>` is copied. I'll be fixing that in a PR soon, which should also then fix this issue as I'll be adding the logic of returning a replayable `BinaryData` in this code path.

Reply:
Thanks Alan.

Would `toReplayableBinaryData()` has implication in the case e.g. the response body be a stream of a few gigabytes? It won't read them all into memory, right?
(we don't have such case so far, but just to be careful)